apis/internal/util.d.ts

```diff
@@ -10,23 +10,8 @@ type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never }
 /** @internal */
 export type XOR<T, U> = (T | U) extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U
 
-// "ArrayLike" is taken since es5, so the underscore is both for @internal and to avoid clashes
-/**
- * A subset of array-like methods, but not `ArrayLike`, as it does not expose `.length`.
- * @see [capire](https://cap.cloud.sap/docs/node.js/cds-reflect#iterable)
- * @internal
- * @since cds 7.9
- */
-export type _ArrayLike<T> = Iterable<T> & {
-  forEach: (handler: (element: T) => any) => void,
-  filter: (predicate: (element: T) => boolean) => Array<T>,
-  map: <R>(converter: (element: T) => R) => Array<R>,
-  some: (predicate: (element: T) => boolean) => boolean,
-  find: (predicate: (element: T) => boolean) => T | undefined,
-}
-
 /**
  * Object structure that exposes both array-like and object-like behaviour.
  * @see [capire](https://cap.cloud.sap/docs/node.js/cds-reflect#iterable)
  */
-export type IterableMap<T> = { [name: string]: T } & _ArrayLike<T>
+export type IterableMap<T> = { [name: string]: T } & Iterable<T>
```

Comment:
```suggestion
export type IterableMap<T> = { [name: string]: T } & Array<T>
```
if I am understanding the change in the implementation correctly, then we probably want `Array` here to still allow `.reduce`, `.map`, `.filter`, etc. If we only want a subset thereof or a completely different behaviour, please let me know, so we can look for a more fitting type here.

Reply:
note to self: remove add spread in tests

Reply:
nvm, I saw now that the methods are actually gone in the implementation.